### ISSUE TITLE
Could this tool be used to auto-format tsql codebase?

### ISSUE THREAD

**Issue by kabirbaidhya** [tsqllint/tsqllint#254]
We need to reformat our sql files consistently both from our editor and check it in the CI. Could it be used for that purpose?

**Comment by nathan-boyd** [comment]
FWIW @stephenweaver has been adding exactly this functionality to the main app. For plugins nothing actually prevents plugins from updating the files, once that feature is stable in the main app it would made sense to explicitly support plugins making updates by supplying a file stream so something so we dont get into multiple write situations. we'd also need to establish some sort of hierarchy like plugins write last or something along those lines.

**Comment by stephenweaver** [comment]
@aharpervc  As part of the latest few changes, I've added an interface called `IPluginWithRules` that inherits from the original `IPlugin` interface that's been around for years. If you your dll has a class the implements this new interface and you provide the required `Dictionary<string, ISqlLintRule> Rules`, then in your rule classes that implement `ISqlLintRule` can define a method called  `FixViolation` that essentially gives you an event hook to fix violations the same way this project does it. 

I've added it to my company's plugin, and it works well for our needs. For brand new plugins there's probably some more features that could be added to make the plugin development experience a little simpler, but this was what I came up with that required minimal changes to existing plugins (that used the `IPlugin` interface) that wanted to add support for the `--fix` feature.

**Comment by aharpervc** [comment]
Interesting. From searching the code, it looks like a plugin's FixViolation implementation adds FileLineActions, and then the framework can execute those FileLineActions. Is that about right?

**Comment by stephenweaver** [comment]
Yep! exactly. There's also `FixHelpers` class that I added to the common project to make is easier to work with. Specifically the `FindViolatingNode`, `FindNodes`, and `GetString` methods I ended up using a lot. I'd suggest looking through some of the rules defined in this repo to see how they can be used. At some point in the next few weeks, I'll write up a PR to update how to write plugins with an example that include FixViolation, but there's no documentation like the code itself :P
